docs/adr/ADR-005-User_attribution_immutable_state_records.md

```diff
@@ -0,0 +1,197 @@
+
+**Pros:**
+
+- No additional fields needed on models
+- Reuses existing audit infrastructure
+- Complete historical context is available
+
+**Cons:**
+
+- **Violates ADR-004 principle**: ADR-004 explicitly decided that audit logs should not be used for functional purposes
+- **Coupling risk**: Application functionality would break if audit logs are archived, pruned, or moved to cold storage
+- **Performance concerns**: Querying audit logs for functional UI operations adds unnecessary overhead
+- **Separation of concerns**: Mixes security/compliance data with functional application data
+- **Different lifecycle**: Audit data may have different retention policies than functional data
+
+#### 3. Create separate attribution tables
+
+Create dedicated attribution tables (e.g., `AppointmentStatusAttribution`) that track user actions separate from both the main models and audit logs.
+
+**Pros:**
+
+- Clear separation between functional data, attribution, and audit
+- Flexible schema for attribution-specific fields (reason, notes, etc.)
+- Could support multiple attribution records per state (e.g., "created by" and "approved by")
+
+**Cons:**
+
+- **Significant over-engineering**: Creates extra complexity for a simple need
+- **Additional joins**: Requires extra database joins for every query that needs attribution
+- **More tables to maintain**: Additional models, migrations, and schema complexity
+- **Violates YAGNI**: We don't have requirements for multiple attributions, approval chains, or attribution-specific metadata
+- **Harder to query**: Simple queries like "show me this status and who created it" require joins
+
+#### 4. Only add user_id to immutable state records (chosen approach)
+
+Add `user_id` directly to immutable state record models like `AppointmentStatus`.
+
+**Pros:**
+
+- **Pragmatic and minimal**: Solves the actual requirement without over-engineering
+- **Perfect fit for immutable records**: These records are never updated, so user attribution is complete and never lost
+- **Simple queries**: No additional joins needed - the attribution is right on the record
+- **Natural pattern**: Each state transition includes "what" (the state), "when" (timestamp), and "who" (user)
+- **YAGNI-compliant**: Only builds what we need now, doesn't add unused infrastructure
+- **Scalable pattern**: Easy to apply the same approach to future state-based workflows
+- **Clear boundaries**: Functional attribution is separate from audit logs, mutable records use different approaches when needed
+
+**Cons:**
+
+- **Pattern inconsistency**: Different models have different attribution approaches (though this reflects different data characteristics)
+- **Future decision needed**: When we need attribution on mutable records, we'll need to decide on an approach (though this is appropriate - we can make that decision when we understand the requirements)
+
+### Rationale
+
```

Comment:
I'm in favour of shorter ADRs. Can we drop this section or summarise in a paragraph? (It mostly reiterates the pros and cons above)